src/PlatformAdapter/Native/Exec/NativeExecImpl.cpp

```diff
@@ -0,0 +1,253 @@
+#include "NativeExecImpl.hpp"
+
+#include "Utils/Utils.hpp"
+
+#undef INFO
+#define INFO(msg, ...)
+
+#undef DEBUG
+#define DEBUG(msg, ...)
+
+/*
+ * NativeExecImpl
+ */
+bool NativeExecImpl::schedule_eval(const std::string& code)
+{
+  if (m_state == DEAD)
+  {
+    FAILED("#NativeExecImpl::schedule_eval, error, dead state");
+    return false;
+  }
+
+  while (m_state != RUNNING)
+  {
+  }
+
+  NativeEvalTask* task = new NativeEvalTask();
+  task->m_code = code;
+  task->m_exec_impl_ptr = this;
+
+  const std::lock_guard<std::mutex> lock(m_tasks_mutex);
+  m_tasks.insert(task);
```

Comment:
could this mutex lock be released immediately after insert? Releasing it after uv invokes may be not good.

Reply:
Yes, updated.